newIDE/app/src/ResourcesList/LocalPiskelBridge.js

```diff
@@ -53,9 +53,9 @@ export const openPiskel = ({
   ipcRenderer.on(
     'piskel-changes-saved',
     (event, outputResources, newAnimationName, receivedData) => {
-      const externalEditorData = {
+      const externalEditorData = receivedData.data ? {
```

Comment:
I'm really not sure about the usage of an empty object `{}`.
99% of the case when it's used, it's to cover up a problem that could be fixed by using null and testing properly if the object is null or not.

I think things can be simplified by naming also properly `externalEditorData` to `metadata` here. As far as I can see, this object will contain something called `pskl`. Which is exactly what we save in metadata. (there is even a comment telling "In case there are multiple frames, pass back the metadata to the editor and let it store it at an appropriate place.". So externalEditorData here IS metadata).

So we can do:

```
(event, outputResources, newAnimationName, receivedData) => {
const metadata = receivedData.data ? { pskl: receivedData } : null;

// ...

if (receivedData.singleFrame) {
  if (metadata) {
          resourcesManager
          .getResource(path.relative(projectPath, outputResources[0].path))
          .setMetadata(JSON.stringify(metadata));	         
  } 
} else {
   //...
  onChangesSaved(outputResources, newAnimationName, metadata);
}
```

null must be understand as "there is no metadata to save". Using an empty object and testing for pskl inside it looks like a workaround. What do you think?

I know it's like I'm nitpicking on useless stuff, but I would like to get this right and have to most simplified solution. Otherwise we'll miss another edge case.

Reply:
I agree its clearer this way, but was wondering if using null might break the flow check. I can add a ? before Object if required

Reply:
in another review note you asked me to rename metadata to externalEditorData :D

Reply:
> I agree its clearer this way, but was wondering if using null might break the flow check. I can add a ? before Object if required

Yes that's the good thing to do, make this a `?Object` to tell that "here we can pass metadata or something that is null to specify that there is no metadata to store along with the resource/direction".